### DISCUSSION TITLE
recommended way to use as reusable form component

### DISCUSSION THREAD

**Discussion by niralparmar** [react-hook-form/react-hook-form#2711] [Q&A] [answered]
Hello,
i am building a component lib for our org, part of that, i am planning to build a reusable form component based of react-hook-form.
idea here would be hide all the complexity into a wrapper component such as register, watch and other stuff. consumer of the component should not worry about importing those function. wrapper component will take care of "regsiter" ,"watch" , validation rules etc.
here is the usage i am thinking 

`import { OrgForm }  from '@org/component/form'`

`exportdefault function App() {` 
 `  const onSubmit = (data) => console.log(data);` 
  ` return (` 
      ` <OrgForm onSubmit={onSubmit}>` 
       `  <div className="form-group">` 
          ` <Input name="firstName" />` 
        ` </div>` 
       `  <div className="form-group">` 
          ` <Input name="lastName" />` 
        ` </div>` 
        ` <div className="form-group">` 
          ` <Select name="sex" options={["female", "male"]} />` 
        ` </div>` 

        ` <Input type="submit" value="Submit" />` 
      ` </OrgForm>` 
  ` );` 
` }` 

Smartform strategy wont work here as inputs are not at the root level and we have many deep nested inputs such as fieldsets/tabs within tabs/cascade drop downs .

any idea?

thank you.

**Comment by bluebill1049** [comment, marked as answer]
You can use ConnectedForm: https://react-hook-form.com/advanced-usage#ConnectForm
